Bootstrap may failed with a nil pointer error if download of a component fails

# Bug Report

## Description

## Expected behavior

In case of download error a clear error message should be returned.
Like bellow.

```
Starting a local Yorc server to bootstrap a remote Yorc server...
...local Yorc server started

Downloading https://fastconnect.org/maven/content/repositories/opensource/alien4cloud/alien4cloud-dist/2.2.0/alien4cloud-dist-2.2.0-dist.tar.gz to work/alien4cloud-dist.tar.gz
Bootstrap error Get https://fastconnect.org/maven/content/repositories/opensource/alien4cloud/alien4cloud-dist/2.2.0/alien4cloud-dist-2.2.0-dist.tar.gz: x509: certificate has expired or is not yet valid
```

## Actual behavior

```
Starting Consul (logs: work/consul.log)..... Consul started!
Starting a local Yorc server to bootstrap a remote Yorc server...
...local Yorc server started
Downloading https://fastconnect.org/maven/content/repositories/opensource/alien4cloud/alien4cloud-dist/2.2.0/alien4cloud-dist-2.2.0-dist.tar.gz to work/alien4cloud-dist.tar.gz

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x14dda2a]

goroutine 1 [running]:
github.com/ystia/yorc/v4/commands/bootstrap.download(0xc00001e240, 0x83, 0x18c80a8, 0x17, 0x18ac9cf, 0x4, 0x1, 0x0, 0x0, 0x0, ...)
	/home/jenkins/agent/workspace/yorc-ci-oss-it-light_develop/commands/bootstrap/setup.go:492 +0x2fa
github.com/ystia/yorc/v4/commands/bootstrap.deployTopology(0x18ac9cf, 0x4, 0xc0002d2a60, 0x20, 0x0, 0x0)
	/home/jenkins/agent/workspace/yorc-ci-oss-it-light_develop/commands/bootstrap/deployment.go:37 +0xac
github.com/ystia/yorc/v4/commands/bootstrap.bootstrap(0xc0004dbd28, 0x5679e4)
	/home/jenkins/agent/workspace/yorc-ci-oss-it-light_develop/commands/bootstrap/bootstrap.go:183 +0x3e3
github.com/ystia/yorc/v4/commands/bootstrap.init.0.func1(0xc000394840, 0xc0002ebb20, 0x0, 0x2)
	/home/jenkins/agent/workspace/yorc-ci-oss-it-light_develop/commands/bootstrap/bootstrap.go:51 +0x26
github.com/spf13/cobra.(*Command).execute(0xc000394840, 0xc0002ebac0, 0x2, 0x2, 0xc000394840, 0xc0002ebac0)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.6/command.go:844 +0x2aa
github.com/spf13/cobra.(*Command).ExecuteC(0x26b89c0, 0xc0004dbf20, 0x43c4ea, 0x264da80)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.6/command.go:945 +0x317
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.6/command.go:885
main.main()
	/home/jenkins/agent/workspace/yorc-ci-oss-it-light_develop/main.go:29 +0x31
script returned exit code 2
```

## Steps to reproduce the issue

1. Start a bootstrap
2. Simulate a network error when downloading

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,10 @@
 
 ## UNRELEASED
 
+### BUG FIXES
+
+* Bootstrap may failed with a nil pointer error if download of a component fails ([GH-634](https://github.com/ystia/yorc/issues/634))
+
 ## 4.0.0 (April 17, 2020)
 
 ### BUG FIXES

--- a/commands/bootstrap/setup.go
+++ b/commands/bootstrap/setup.go
@@ -487,16 +487,15 @@ func download(url, fileName, destinationPath string, overwrite bool) (string, er
 		defer outputFile.Close()
 
 		response, err := http.Get(url)
-
-		//bar
-		bar := pb.Full.Start(int(response.ContentLength))
-		barReader := bar.NewProxyReader(response.Body)
-
 		if err != nil {
 			outputFile.Close()
 			return filePath, err
 		}
 		defer response.Body.Close()
+
+		//bar
+		bar := pb.Full.Start(int(response.ContentLength))
+		barReader := bar.NewProxyReader(response.Body)
 
 		_, err = io.Copy(outputFile, barReader)
 		outputFile.Close()

